In [3]:
import math

a,b,c=[],[],[]

a=[[1,6],[3,10],[7,4]]
b=[[1,3],[0.5,2]]
c=[[2,-2],[-3,1]]

def inicilizar_Matriz(fil,col):
    
  
    mat= [None] * fil
    
    if(fil>0):
        for i in range(fil):
            mat[i] = [None] * col
            
    return mat

def imprimir_matriz(a):
    
    j=0
    
    while j<len(a):
        print(a[j])
        j+=1

def multiplicacion_matrices(a,b):
    
    c=[]
    
    fil_a,col_a=len(a),len(a[0])
    fil_b,col_b=len(b),len(b[0])
    
    i,j,w,x,y,aux=0,0,0,0,0,0
    
    #print(fil_a,col_a,fil_b,col_b)

    if(col_a==fil_b):
        
        c=inicilizar_Matriz(fil_a,col_b)
            
        if(fil_a>col_b):
            x=fil_a
        else:
            x=col_b
        
        while i<x:
            
            while w<col_b:
                while j<fil_b:
                    aux+=a[y][j]*b[j][w]
                    j+=1
                j=0
                   
                c[y][w]=aux
             
                aux=0
                w+=1
            w=0
            if(y<fil_a):
                y+=1
            
            i+=1
    else:
        print("No es posible multiplicar las dos matrices")
    
    return c

print("(a)")
print("AxB") 
imprimir_matriz(multiplicacion_matrices(a,b))
print("AxC") 
imprimir_matriz(multiplicacion_matrices(a,c))
print("BxC") 
imprimir_matriz(multiplicacion_matrices(b,c))
print("CxB") 
imprimir_matriz(multiplicacion_matrices(c,b))


(a)
AxB
[4.0, 15]
[8.0, 29]
[9.0, 29]
AxC
[-16, 4]
[-24, 4]
[2, -10]
BxC
[-7, 1]
[-5.0, 1.0]
CxB
[1.0, 2]
[-2.5, -7]


a) Ejecute todas las multiplicaciones que sea posible calcular
entre parejas de las matrices.
b) Utilice el método del recuadro PT3.2 para justificar por qué
no se puede multiplicar a las demás parejas.
c) Emplee los resultado del inciso a) para ilustrar por qué es
importante el orden de la multiplicación.

Solucion

a)
AxB
[4.0, 15]
[8.0, 29]
[9.0, 29]
AxC
[-16, 4]
[-24, 4]
[2, -10]
BxC
[-7, 1]
[-5.0, 1.0]
CxB
[1.0, 2]
[-2.5, -7]
b)Para que dos matrices se puedan multiplicar (AxB) el numero de columnas de A debe ser igual a el numero filas de B
c)la multiplicacion entre matrices no es conmutativa